### ISSUE TITLE
Query with a not-equals clause converts OR statements to AND

### ISSUE THREAD

**Issue by Farthal** [chriseldredge/Lucene.Net.Linq#77]
If I try to run a query such as

`session.Query().Where(movie => movie.Year == 2010 || movie.Title != "Interstellar")`

the results are equivalent to joining the clauses with &&.  The || statement and != clause work fine if they aren't used together in a single query.

**Comment by chriseldredge** [comment]
Sounds like a bug. It should produce a Lucene query like

```
Year:2010 (*:* -Title:Interstellar)
```

I'll have to look at the code to see what actual query is being produced.

**Comment by chriseldredge** [comment]
The actual query produced is

```
Scalar:2010 -Name:interstellar
```

This is effectively a boolean AND instead of OR.

**Comment by chriseldredge** [comment]
Fixed in fa0478721a14601dada12115ed9363f17ddd3a75.

**Comment by chriseldredge** [comment]
Released in https://github.com/themotleyfool/Lucene.Net.Linq/releases/tag/v3.5.2
